crates/rune/src/macros/functions.rs

```diff
@@ -70,3 +71,20 @@ where
 {
     current_context(|ctx| ctx.stringify(stream).to_string())
 }
+
+/// Parse the given input as the given type that implements
+/// [Parse][crate::parsing::Parse].
+///
+/// # Panics
+///
+/// This will panic if it's called outside of a macro context.
+pub fn parse_all<T>(source: &str) -> Result<T, ParseError>
+where
+    T: Parse,
+{
+    current_context_mut(|ctx| {
+        let sources = ctx.sources_mut();
+        let source_id = sources.insert(Source::new("macro", source));
+        crate::parse_all(source, source_id)
+    })
+}
```

Comment:
This is the `parse_all` implementation that can now be implemented inside of macros as a result of this change.